twisted/conch/ssh/transport.py

```diff
@@ -1499,6 +1598,59 @@ def ssh_KEXINIT(self, packet):
                     self._dhMaximalGroupSize,
                     ))
 
+    def _ssh_KEX_ECDH_REPLY(self, packet):
+        """
+        Called to handle a reply to a ECDH exchange message(KEX_ECDH_INIT).
+
+        Like the handler for I{KEXDH_INIT}, this message type has an
+        overlapping value.  This method is called from C{ssh_KEX_DH_GEX_GROUP}
+        if that method detects a non-group key exchange is in progress.
+
+        Payload::
+
+            string serverHostKey
+            string server Elliptic Curve Diffie-Hellman public key
+            string signature
+
+        We verify the host key and continue if it passes verificiation.
+        Otherwise raise an exception and return.
+
+        @type packet: L{bytes}
+        @param packet: The message data.
+
+        @return: None.
+        """
+        #Get the host public key
+        self.theirECHost, packet = getNS(packet)
+
+        #Get the raw ECDH public key bytes
+        pktPub, packet = getNS(packet)
+
+        #Get the signature
+        signature, packet = getNS(packet)
+
+        h = _kex.getHashProcessor(self.kexAlg)()
+        h.update(NS(self.ourVersionString))
+        h.update(NS(self.otherVersionString))
+        h.update(NS(self.ourKexInitPayload))
+        h.update(NS(self.otherKexInitPayload))
+        h.update(NS(self.theirECHost))
+
+        #Take the provided public key and transform it into a format for the cryptography module
+        self.theirECPub = ec.EllipticCurvePublicNumbers.from_encoded_point(self.curve, pktPub).public_key(default_backend())
```

Comment:
Minor, optional nitpick: it seems odd to me that we stop to load the curve mid-way through computing the hash (especially to do a side-effect).  I'd find this method easier to read if the side-effects (all the `self.x = y`) were isolated to one place - ideally at the _end_ of the method, after we have finished validating everything we're going to load.  I confess I don't fully understand the consequences of saving `theirECPub` and `theirECHost` and then raising `InvalidSignature` before calling `_keySetup`, but in general this is a recipe for bugs.